### PR TITLE
feat(model parallelism): move labels to the same device as logits for M2M100

### LINKED ISSUE #22561: Make all Transformer models compatible with model parallelism
Accelerate makes it easy to load a model on multiple GPUs with `device_map="auto"`. This in turn allows users to train model with naive model parallelism if they have several GPUs.

A problem that happens in Transformers, with model with heads (so not XxxModel but for instance XxxModelForSequenceClassification) is that the labels end up on a different device than the logits and there is a device mistmatch error.

Thankfully, there is an easy fix for that! #22535 shows how to fix this for T5 by just moving the labels to the same device as the logits they are compared to. This is a noop when the devices are the same, and fixes the issue if devices are different.

We would like help from the community to extend this to all models that support model parallelism, which are:
- [x] BART
- [x] BigBirdPegasus
- [x] BLIP2
- [x] BLOOM
- [x] BridgeTower
- [x] CamemBERT
- [ ] CLIP
- [ ] CLIPSeg
- [ ] CodeGen
- [x] Data2Vec Text
- [x] Deit
- [x] ESM
- [x] GPT-2
- [x] GPT-Neo
- [x] GPT-NeoX
- [x] GPT-NeoX Japanese
- [ ] GPT-J
- [x] GPT-San
- [ ] JukeBox
- [ ] Lilt
- [x] LLaMA (`LlamaForSequenceClassification` only)
- [x] Longformer
- [x] LongT5
- [ ] Luke
- [ ] M2M100
- [x] mBART
- [x] mT5
- [ ] NLLB
- [ ] OPT
- [ ] Owl-ViT
- [ ] Pix2Struct
- [x] PLBART
- [x] RoBERTa
- [x] RoBERTa PreLayerNorm
- [ ] SwitchTransformer
- [x] T5
- [ ] Vilt
- [x] ViT
- [x] ViT-Hybrid
- [ ] Whisper
- [x] XLM-RoBERTa

If you would like to grab one of those models and apply the same fix as #22535 to all the model with heads, please leave a comment here!

### PR DESCRIPTION
**Pull request by elabongaatuo**
# What does this PR do?

Moves labels to same device as logits for M2M100

Related to  #22561

@sgugger hello please review.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Comment by elabongaatuo** [comment]
> Thanks a lot!

Thank you 😊
